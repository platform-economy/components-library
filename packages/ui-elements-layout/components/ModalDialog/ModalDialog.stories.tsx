import * as React    from 'react';
import { storiesOf } from '@storybook/react';
import { ModalDialog } from './index';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faTimes } from '@fortawesome/free-solid-svg-icons';
import styled from 'styled-components';
import { boolean, text } from '@storybook/addon-knobs';

import { actionWithoutEvent } from '../../../../helpers/storybook-helpers';

library.add(
  faTimes,
);

export const Title = styled.h1`
  padding-top: 12px;
  font-size: 1.5em;
  text-align: center;
  color: white;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.33;
  letter-spacing: -0.43px;
  text-align: center;
  color: ${props => props.theme.palette.text};
`;

export const Subtitle = styled.h2`
  font-size: .875em;
  font-weight: normal;
  color: white;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.71;
  letter-spacing: -0.43px;
  text-align: center;
  margin-top: -3px;
  grid-column-start: 2;
  grid-column-end: 2;
  grid-row-start: 2;
  grid-row-end: 2;
  color: ${props => props.theme.palette.text};
`;

storiesOf('ui-elements-layout/ModalDialog', module)
  .add('ModalDialog', (() => (
      <ModalDialog
        closeIcon={
          <FontAwesomeIcon icon="times" />
        }
        onCloseClick={actionWithoutEvent('clicked-close-button')}
        header={
          [
            <Title key="title">Title</Title>,
            <Subtitle key="subtitle">Subtitle</Subtitle>,
          ]
        }
      >
        content
      </ModalDialog>

  )))
  .add('ModalDialog show', () => {

    return(
      <ModalDialog
        closeIcon={boolean('Show close icon', false) ? <FontAwesomeIcon icon="times" /> : undefined}
        header={boolean('Show header', false) ?
          [
            <Title key="example-title">{text('Title', 'Example title')}</Title>,
            <Subtitle key="example-subtitle">{text('Subtitle', 'Example subtitle')}</Subtitle>,
          ] : undefined
        }
      >
        {text('Content', 'Example content')}
      </ModalDialog>
    );
  });
