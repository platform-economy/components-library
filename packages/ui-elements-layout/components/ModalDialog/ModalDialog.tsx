import React, { Component } from 'react';
import styled from 'styled-components';

export type ModalDialogProps = {
  isOpened?: boolean;
  closeIcon?: React.ReactNode;
  onCloseClick?: (e: React.SyntheticEvent) => void;
  header?: React.ReactNode;
  footer?: React.ReactNode;
  children: React.ReactNode;
};

const StyledModalDialog = styled.div`
  width: 680px;
  height: 733px;
  border-radius: 2px;
  box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.5);
  background-color: ${props => props.theme.palette.surface};
  position: relative;
  padding: 10px;
`;

export const CloseIconWrapper = styled.div`
  color: ${props => props.theme.palette.text_inverted};
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.5;
  letter-spacing: -0.43;
  top: 1em;
  position: absolute;
  right: 1.5em;
  &:hover {
      color: ${props => props.theme.palette.active_hovered};
      cursor: pointer;
  }
`;

export class ModalDialog extends Component<ModalDialogProps> {

  render() {

    const { children, onCloseClick, header, closeIcon } = this.props;
    return (
      <StyledModalDialog>
        <header>
            {closeIcon ? (
            <CloseIconWrapper onClick={onCloseClick}>
              {closeIcon}
            </CloseIconWrapper>) : null}
            {header}
        </header>
        <section>
          {children}
        </section>

      </StyledModalDialog>
    );
  }
}
