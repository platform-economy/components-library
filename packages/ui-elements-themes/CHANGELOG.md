# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
