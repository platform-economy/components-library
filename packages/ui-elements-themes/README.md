# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
