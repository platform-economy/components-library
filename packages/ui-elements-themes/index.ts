export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
