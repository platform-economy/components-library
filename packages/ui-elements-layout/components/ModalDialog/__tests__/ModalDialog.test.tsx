import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { Title, Subtitle } from '../ModalDialog.stories';
import { CloseIconWrapper } from '../ModalDialog';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

/*
A set of utilities for testing Styled Components with Jest.
This package improves the snapshot testing experience and
provides a brand new matcher to make expectations on the style rules.
*/
import 'jest-styled-components';

// Component
import { ModalDialog } from '../index';
import { withTestTheme } from '@relayr/ui-elements-themes';

enzyme.configure({ adapter: new Adapter() });

describe('ui-elements-layout', () => {
  describe('Components', () => {
    describe('ModalDialog', () => {

      it('should render with default state', () => {
        const tree = TestRenderer
          .create(
            withTestTheme(
              <ModalDialog
                closeIcon={'Close icon.'}
              >
              {}
              </ModalDialog>,
            ),
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
      it('should render all properties', () => {
        const tree = TestRenderer
          .create(
            withTestTheme(
              <ModalDialog
                closeIcon={'close-icon'}
                header={
                  <>
                  <Title>Title</Title>
                  <Subtitle>Subtitle</Subtitle>
                </>
                }
                footer={'footer'}
              >
              {}
              </ModalDialog>,
            ),
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should render without close icon', () => {
        const tree = TestRenderer
          .create(
            withTestTheme(
              <ModalDialog
              >
              {}
              </ModalDialog>,
            ),
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should handle click event correctly', () => {
        const onCloseClick = jest.fn();
        const tree = enzyme.mount(withTestTheme(
        <ModalDialog
          onCloseClick={onCloseClick}
          closeIcon={'mock-of-close-icon'}
        >
          {}
        </ModalDialog>,
          ));
        tree.find(CloseIconWrapper).simulate('click');
        expect(onCloseClick).toHaveBeenCalledTimes(1);
      });
    });
  });
});
