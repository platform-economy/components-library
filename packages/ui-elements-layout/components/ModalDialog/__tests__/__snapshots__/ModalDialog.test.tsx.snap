// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-layout Components ModalDialog should render all properties 1`] = `
.c0 {
  width: 680px;
  height: 733px;
  border-radius: 2px;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.5);
  background-color: #surface;
  position: relative;
  padding: 10px;
}

.c1 {
  color: #text_inverted;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.5;
  -webkit-letter-spacing: -0.43;
  -moz-letter-spacing: -0.43;
  -ms-letter-spacing: -0.43;
  letter-spacing: -0.43;
  top: 1em;
  position: absolute;
  right: 1.5em;
}

.c1:hover {
  color: #active_hovered;
  cursor: pointer;
}

.c2 {
  padding-top: 12px;
  font-size: 1.5em;
  text-align: center;
  color: white;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.33;
  -webkit-letter-spacing: -0.43px;
  -moz-letter-spacing: -0.43px;
  -ms-letter-spacing: -0.43px;
  letter-spacing: -0.43px;
  text-align: center;
  color: #text;
}

.c3 {
  font-size: .875em;
  font-weight: normal;
  color: white;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.71;
  -webkit-letter-spacing: -0.43px;
  -moz-letter-spacing: -0.43px;
  -ms-letter-spacing: -0.43px;
  letter-spacing: -0.43px;
  text-align: center;
  margin-top: -3px;
  grid-column-start: 2;
  grid-column-end: 2;
  grid-row-start: 2;
  grid-row-end: 2;
  color: #text;
}

<div
  className="c0"
>
  <header>
    <div
      className="c1"
    >
      close-icon
    </div>
    <h1
      className="c2"
    >
      Title
    </h1>
    <h2
      className="c3"
    >
      Subtitle
    </h2>
  </header>
  <section />
</div>
`;

exports[`ui-elements-layout Components ModalDialog should render with default state 1`] = `
.c0 {
  width: 680px;
  height: 733px;
  border-radius: 2px;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.5);
  background-color: #surface;
  position: relative;
  padding: 10px;
}

.c1 {
  color: #text_inverted;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.5;
  -webkit-letter-spacing: -0.43;
  -moz-letter-spacing: -0.43;
  -ms-letter-spacing: -0.43;
  letter-spacing: -0.43;
  top: 1em;
  position: absolute;
  right: 1.5em;
}

.c1:hover {
  color: #active_hovered;
  cursor: pointer;
}

<div
  className="c0"
>
  <header>
    <div
      className="c1"
    >
      Close icon.
    </div>
  </header>
  <section />
</div>
`;

exports[`ui-elements-layout Components ModalDialog should render without close icon 1`] = `
.c0 {
  width: 680px;
  height: 733px;
  border-radius: 2px;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.5);
  background-color: #surface;
  position: relative;
  padding: 10px;
}

<div
  className="c0"
>
  <header />
  <section />
</div>
`;
